mypyc/irbuild/function.py

```diff
@@ -313,6 +315,20 @@ def c() -> None:
     # calculate them *once* when the function definition is evaluated.
     calculate_arg_defaults(builder, fn_info, func_reg, symtable)
 
+    if is_singledispatch:
+        # add the generated main singledispatch function
+        builder.functions.append(func_ir)
+        # create a dispatch function (a function that checks the first argument type and dispatches
+        # to the correct implementation)
+        builder.enter()
+        assert isinstance(fitem, FuncDef)
+        generate_singledispatch_dispatch_function(builder, fn_info.name, fitem)
+        args, _, blocks, _, fn_info = builder.leave()
+        dispatch_name = decorator_helper_name(name) if is_decorated else name
+        func_decl = FuncDecl(dispatch_name, None, builder.module_name, sig)
+        dispatch_func_ir = FuncIR(func_decl, args, blocks)
+        return dispatch_func_ir, None
```

Comment:
Do you think this would be cleaner if we pulled it out into another function? (The answer could be no)

Reply:
It probably would be cleaner. I'll move it into another function.